### PR TITLE
Feature add circleci compatibility

### PR DESCRIPTION
**Pull request by rafaelje**
Add compatibility to circleci with all this task:

- update version code 
- update version name 
- generate changelog 
- build 
- instrumented test
- deploy with fastlane beta and production rollout
- deploy with github releaser
- generate documentation with javadoc
- generate documentation code coverage
- generate documentation android test
- publish changelog on gh-pages
- publish documentation on gh-pages

close: https://github.com/flyve-mdm/flyve-mdm-android-inventory-agent/issues/32

### CODE REVIEW

**Comment by ajsb85** [comment]
@rafaelje Hi, @Rafaelje 
The names of the tests: test_16 and test_25 are the Android API versions?

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 version: 2
 jobs:
   build:
-    working_directory: ~/code
+    working_directory: ~/flyve
```
Comment (ajsb85) [inline .circleci/config.yml]: Keep in mind, the registered name is: Flyve MDM
You can use: flyve-mdm or flyve_mdm as slug.

File: .circleci/config.yml [hunk 2]
```diff
@@ -17,12 +18,208 @@ jobs:
...
+          requires:
+            - test_16
+            - test_25
+          filters:
+            branches:
+              only:
+                - develop
+      - documentation:
+          requires:
+            - test_16
```
Comment (ajsb85) [inline .circleci/config.yml]: It's ok?
- test_16
- test_25

Or you need to change to:
- test_api_version_16
- test_api_version_25

File: ci/scripts/ci_updateversion.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+# increment version code, need to be unique to send to store
+./gradlew updateVersionCode -P vCode=$CIRCLE_BUILD_NUM
+
+# increment version on package.json, create tag and commit with changelog
+npm run release -- -m "ci(release): generate **CHANGELOG.md** for version %s"
```
Comment (ajsb85) [inline ci/scripts/ci_updateversion.sh]: I dont like Markdown format in the commit messages

File: .circleci/config.yml (hunk 2 above)
Comment (rafaelje) [reply .circleci/config.yml]: Ready @ajsb85 you are right, now is test_api_version_16 and test_api_version_25

File: ci/scripts/ci_updateversion.sh (hunk 3 above)
Comment (rafaelje) [reply ci/scripts/ci_updateversion.sh]: Ready remove markdown with plain text

**Comment by ajsb85** [comment]
Ready to merge?

**Pull request merged**